docs/reference/aggregations/bucket/variablewidthhistogram-aggregation.asciidoc

```diff
@@ -0,0 +1,85 @@
+[[search-aggregations-bucket-variablewidthhistogram-aggregation]]
+=== Variable Width Histogram Aggregation
+
+This is a multi-bucket aggregation similar to <<search-aggregations-bucket-histogram-aggregation>>.
+However, the width of each bucket is not specified. Rather, a target number of buckets is provided and bucket intervals
+are dynamically determined based on the document distribution. This is done using a simple one-pass document clustering algorithm
+that aims to obtain low distances between bucket centroids. Unlike other multi-bucket aggregations, the intervals will not
+necessarily have a uniform width.
+
+TIP: The number of buckets returned will always be less than or equal to the target number.
+
+Requesting a target of 3 buckets.
+
+[source,console]
+--------------------------------------------------
+POST /sales/_search?size=0
+{
+    "aggs" : {
+        "prices" : {
+            "variable_width_histogram" : {
+                "field" : "price",
+                "buckets" : 2
+            }
+        }
+    }
+}
+--------------------------------------------------
+// TEST[setup:sales]
+
+Response:
+
+[source,console-result]
+--------------------------------------------------
+{
+    ...
+    "aggregations": {
+        "prices" : {
+            "buckets": [
+                {
+                    "key": 10.0,
+                    "doc_count": 2
+                },
+                {
+                    "key": 150.0,
+                    "doc_count": 5
+                }
+            ]
+        }
+    }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": $body.took,"timed_out": false,"_shards": $body._shards,"hits": $body.hits,/]
+
+==== Shard size
```

Comment:
I think these should be below the description of the algorithm. That description is short and pretty important to understand.